src/ride/vehicle.c

```diff
@@ -5381,11 +5381,58 @@ static void vehicle_update_track_motion_up_stop_check(rct_vehicle *vehicle)
 	}
 }
 
+void apply_block_section_stop_site(rct_vehicle *vehicle, rct_ride *ride){
+	int trackType = vehicle->track_type >> 2;
+
+	rct_map_element *trackElement =  map_get_track_element_at_of_type(
+		vehicle->track_x,
+		vehicle->track_y,
+		vehicle->track_z >> 3,
+		trackType
+	);
+
+	if (trackType == TRACK_ELEM_END_STATION) {
+		if (trackElement->flags & MAP_ELEMENT_FLAG_BLOCK_BREAK_CLOSED) {
+			RCT2_GLOBAL(0x00F64E18, uint32) |= VEHICLE_UPDATE_MOTION_TRACK_FLAG_10;
+		}
+	} else {
+		// If the track piece is a block section stop site
+		if (trackType == TRACK_ELEM_CABLE_LIFT_HILL || trackType == TRACK_ELEM_BLOCK_BRAKES || track_element_is_lift_hill(trackElement)) {
+			// If the site is in a "train blocking" state
+			if ((trackElement->flags & MAP_ELEMENT_FLAG_BLOCK_BREAK_CLOSED) && ride_is_block_sectioned(ride)) {
+				// Slow it down till completely stop the car
+				RCT2_GLOBAL(0x00F64E18, uint32) |= VEHICLE_UPDATE_MOTION_TRACK_FLAG_10;
+				vehicle->acceleration = 0;
+				// If the vehicle is slow enough, stop it. If not, slow it down
+				if (vehicle->velocity <= 0x20000) {
+					vehicle->velocity = 0;
+				} else {
+					vehicle->velocity -= vehicle->velocity >> 3;
+				}
+			} else {
+				// If the site is a block brake
+				if (trackType == TRACK_ELEM_BLOCK_BRAKES && vehicle->velocity >= 0) {
```

Comment:
If you changed this bit of code into a separate function you could make continuous circuit not call a function called block section mode